buildSrc/src/main/java/fmlbuild/RunConfigurationsPlugin.java

```diff
@@ -0,0 +1,167 @@
+                spec.shouldResolveConsistentlyWith(runtimeClasspathConfig);
+            });
+            runtimeModulesConfig.fromDependencyCollector(runConfiguration.getDependencies().getModulepath());
+
+
+            project.getTasks().create("run" + capitalizedName, JavaExec.class, task -> {
+                task.classpath(sourceSet.getRuntimeClasspath());
+                task.getMainClass().set(runConfiguration.getMainClass());
+                var jvmArguments = task.getJvmArguments();
+                jvmArguments.addAll(runConfiguration.getJvmArguments());
+                jvmArguments.addAll(runConfiguration.getSystemProperties().map(properties -> {
+                    return properties.entrySet().stream().map(entry -> "-D" + entry.getKey() + "=" + entry.getValue()).toList();
+                }));
+                jvmArguments.addAll(runtimeModulesConfig.getElements().map(elements -> {
+                    if (elements.isEmpty()) {
+                        return List.of();
+                    }
+                    return List.of("-p", elements.stream()
+                            .map(FileSystemLocation::getAsFile)
+                            .map(File::getAbsolutePath)
+                            .collect(Collectors.joining(File.pathSeparator))
+                    );
+                }));
+
+                // I don't see a way to avoid querying this provider eagerly...
+                project.afterEvaluate(ignored -> {
+                    task.setGroup(runConfiguration.getTaskGroup().get());
+                });
+                // Use the project java version to launch
+                task.getJavaLauncher().set(getJavaToolchainService().launcherFor(javaSpec -> {
+                    javaSpec.getLanguageVersion().set(java.getToolchain().getLanguageVersion());
+                }));
+
+                // We need to pass some inputs to the task that cannot be set via Properties yet
+                var taskInputs = task.getInputs();
+                // To avoid accidentally tripping Gradle by passing a "File", we convert it to String in the provider
+                taskInputs.property("runWorkingDirectory", runConfiguration.getWorkingDirectory().map(Directory::getAsFile).map(File::getAbsolutePath));
+                taskInputs.property("runProgramArgs", runConfiguration.getProgramArguments());
+                task.doFirst(RunConfigurationsPlugin::configureJavaExec);
+            });
+        });
+        runConfigurations.whenObjectRemoved(installation -> {
+            throw new GradleException("Cannot remove installations once they have been registered");
+        });
+
+        project.afterEvaluate(ignored -> {
+            var ijRunConfigs = getIntelliJRunConfigurations(project);
+            if (ijRunConfigs == null) {
+                return;
+            }
+
+            for (var settings : runConfigurations) {
```

Comment:
Currently these runs are outright broken anyway. We should maybe add Gradle task runs directly.